ops/prod/api.tf

```diff
@@ -68,3 +68,38 @@ module "report_stream_reporting_functions" {
   environment = local.env
   tenant_id   = data.azurerm_client_config.current.tenant_id
 }
+
+resource "azurerm_monitor_autoscale_setting" "prod_autoscale" {
```

Comment:
I know this is late in the game but I think I'd prefer a `module` here, as we may switch other envs to also use autoscale settings. I'm not sure if the ticket captures this scope but we should probably also be using this resource in every env, rather than just prod. If we don't, this isn't getting tested via either a `plan` or `apply` until prod. We don't need special weekend settings or anything - just setting the autoscale settings to mimic the fixed size settings we have now would be enough.

Reply:
I see where you're coming from on this, and see the validity of what you're proposing! The ticket is a little vague, but I think your request is beyond the scope of this PR. The initial goal was to encompass the existing settings, and given the variability between environments, I intentionally left things as a single resource at this time.

I do see your concern regarding environmental testing. If we're making an emphasis on quality moving forward, we should absolutely work to standardize across environments, and ensure we run deployment tests on non-prod. 

Given all of that...I want to go over this at the Eng Sync this week. I'll add it to the agenda, and we can briefly discuss what kind of scaling rules we would like in our lower environments, whether fixed or variable. Once we have that information, we can reprioritize this with your module addition added.

Let's get all the pieces in place, so we only have to touch this once. 😄

Reply:
This was actually a suggestion to be able to do just that! If the plan is to ever add this to more envs later, we may as well use a `module` from the start to avoid a later `terraform state mv`. We can still capture that other work somewhere else, as I agree that's scope creep on this issue.

Reply:
I just approved, since this probably isn't worth holding things up over

Reply:
I can't unhear your excellent explanation, though! A module from the start would prevent the `state` call, and I don't think it will add terribly much additional work. I will just need to shuffle this behind our JVM heap item, since our audit log stuff is a bit more pressing. I think I missed the significance of the initial ask when I first worked this up weeks ago, and I want to make sure it is properly addressed.

Reply:
Alright, module conversion complete. I ran into a bit of trouble deconstructing the profile blocks to a point where they could be easily augmented...but, that isn't an immediate need, since we don't have any additional scaling asks at this time. I have captured that effort in this ticket: https://github.com/cdcgov/prime-simplereport/issues/3526 If we need to prioritize dynamic autoscaling, we certainly can, and I can revisit this immediately!